### PR TITLE
feat: add Type.isTemplateLiteral method

### PR DESCRIPTION
**Pull request by shaharkazaz**
Expose a new `isTemplateLiteral` method from the `Type` class.

No breaking changes were introduced in this PR.

### CODE REVIEW

File: docs/details/types.md [hunk 1]
```diff
@@ -200,25 +200,26 @@ type.isAnonymous();
 type.isAny();
```
Comment (shaharkazaz) [inline docs/details/types.md:200]: I ordered them alphabetically and added `isTemplateLiteral` to the list, that's why you see multiple changes here 🙂

**Review by dsherret — APPROVED** [review]
Also looks good. Thanks!

File: packages/ts-morph/src/compiler/types/Type.ts [hunk 2]
```diff
@@ -389,6 +389,13 @@ export class Type<TType extends ts.Type = ts.Type> {
     return symbol.getName() === "Array" && this.getTypeArguments().length === 1;
   }
 
+    /**
+     * Gets if this is a boolean type.
```
Comment (dsherret) [inline packages/ts-morph/src/compiler/types/Type.ts:393]: template literal. I'll fix it after merge (going to do a release)

Comment (shaharkazaz) [reply packages/ts-morph/src/compiler/types/Type.ts:393]: Missed that 😅

**Pull request merged**